### PR TITLE
sqld: Expose write proxy on primaries

### PR DESCRIPTION
**Pull request by LucioFranco**
This commit exposes the grpc write proxy endpoint on primaries and also refactors how we capture auth. For internal communcation we now send a header instead of encoding the authorization in the grpc message. For external, we go through the regular grpc auth.

### CODE REVIEW

File: sqld/src/auth.rs [hunk 1]
```diff
@@ -119,6 +120,40 @@ impl Auth {
...
 }
 
+impl Authenticated {
+    pub fn from_proxy_grpc<T>(req: &tonic::Request<T>) -> Result<Self, Status> {
+        let auth = match req
+            .metadata()
+            .get(GRPC_PROXY_AUTH_HEADER)
+            .map(|v| v.to_str())
+            .transpose()
+            .map_err(|_| Status::invalid_argument("missing x-proxy-auth"))?
```
Comment (psarna) [inline sqld/src/auth.rs:130]: I think we should be consistent in naming here - is it `x-proxy-auth` or `x-proxy-authorization`?

File: sqld/src/auth.rs [hunk 2]
```diff
@@ -119,6 +120,40 @@ impl Auth {
...
+        {
+            Some("full_access") => Authenticated::Authorized(Authorized::FullAccess),
+            Some("read_only") => Authenticated::Authorized(Authorized::ReadOnly),
+            Some("anonymous") => Authenticated::Anonymous,
+            Some(_) => return Err(Status::permission_denied("invalid authorization level")),
```
Comment (psarna) [inline sqld/src/auth.rs:135]: let's make future debugging easier and print the invalid level as well (`invalid authorization level: {level}`), e.g. when somebody introduces a typo

File: sqld/src/auth.rs [hunk 3]
```diff
@@ -119,6 +120,40 @@ impl Auth {
...
+            None => return Err(Status::invalid_argument("unable to convert to ascii")),
+        };
+
+        Ok(auth)
+    }
+
+    pub fn as_proxy_grpc<T>(&self, req: &mut tonic::Request<T>) {
```
Comment (MarinPostma) [inline sqld/src/auth.rs:142]: nit: The naming could be clearer. The `as_` has a strong connotation in rust, and I would expect to get something back (c.f https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv). Maybe something like `update_gprc_request` would be a better fit?

**Review by MarinPostma — CHANGES_REQUESTED** [review]
This is not enough: embedded replicas are not to be trusted, so we can't trust the authentication header they pass to us. Embedded replicas must forward the JWT to us, so we can actually verify the claims.

**Review by MarinPostma — APPROVED** [review]

File: sqld/src/auth.rs (hunk 3 above)
Comment (LucioFranco) [reply sqld/src/auth.rs:142]: Yeah, I originally changed it because clippy yelled at me, will get a better name.

**Pull request merged**